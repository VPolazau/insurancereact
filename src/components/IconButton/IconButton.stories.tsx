import menuBurger from '../../assets/img/menu_burger.svg';

import { IconButton } from './IconButton';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof IconButton> = {
    title: 'UI/IconButton',
    tags: ['autodocs'],
    component: IconButton,
    argTypes: {
        size: {
            type: 'number',
            description: 'Размер логотипа',
            control: {
                type: 'range',
                min: 20,
                max: 60,
                step: 1,
            },
        },
        icon: {
            type: 'string',
            description: '',
        },
        className: {
            type: 'string',
        },
        alt: {
            type: 'string',
            defaultValue: 'iconButton',
        },
    },
};

export default meta;
type Story = StoryObj<typeof IconButton>;

export const LogoBtn: Story = {
    args: { size: 49, icon: menuBurger },
};
