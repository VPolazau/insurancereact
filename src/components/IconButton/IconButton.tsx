import React from 'react';
import cn from 'classnames';

import styles from './styles.module.css';

interface IIconButton {
    size?: number;
    onClick: (e: React.MouseEvent) => void;
    icon: string;
    className?: string;
    alt?: string;
}

export const IconButton = ({ size = 46, onClick, icon, className, alt = 'iconButton' }: IIconButton) => {
    return (
        <button className={cn(styles.iconButtonBasic, className)} data-testid="iconButtonComponent" onClick={onClick}>
            <img style={{ height: size + 'px' }} src={icon} alt={alt} />
        </button>
    );
};
