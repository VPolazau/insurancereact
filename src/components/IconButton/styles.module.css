.iconButtonBasic {
    display: flex;
    justify-content: center;
    align-items: center;
    align-self: flex-start;
    margin: 0;
    padding: 0;
    border: none;
    background: rgba(255, 255, 255, 0);
    cursor: pointer;
}
