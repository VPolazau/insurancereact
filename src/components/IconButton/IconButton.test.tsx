import { render, fireEvent, screen } from '@testing-library/react';

import menu_burger from '../../assets/img/menu_burger.svg';

import { IconButton } from './IconButton';

describe('IconButton component', () => {
    it('should render the IconButton with default size and trigger onClick event', () => {
        const onClickMock = jest.fn();
        render(<IconButton onClick={onClickMock} icon={menu_burger} />);

        const iconButtonComponent = screen.getByTestId('iconButtonComponent');
        const iconButtonImage = screen.getByAltText('iconButton');

        expect(iconButtonComponent).toBeInTheDocument();
        expect(iconButtonImage).toBeInTheDocument();
        expect(iconButtonImage).toHaveStyle('height: 46px');

        fireEvent.click(iconButtonComponent);
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('should render the IconButton with a custom size and trigger onClick event', () => {
        const onClickMock = jest.fn();
        render(<IconButton size={64} onClick={onClickMock} icon={menu_burger} />);

        const iconButtonComponent = screen.getByTestId('iconButtonComponent');
        const iconButtonImage = screen.getByAltText('iconButton');

        expect(iconButtonComponent).toBeInTheDocument();
        expect(iconButtonImage).toBeInTheDocument();
        expect(iconButtonImage).toHaveStyle('height: 64px');

        fireEvent.click(iconButtonComponent);
        expect(onClickMock).toHaveBeenCalledTimes(1);
    });

    it('should match snapshot for default size', () => {
        const onClickMock = jest.fn();
        const { container } = render(<IconButton onClick={onClickMock} icon={menu_burger} />);
        expect(container).toMatchSnapshot();
    });

    it('should match snapshot for custom size', () => {
        const onClickMock = jest.fn();
        const { container } = render(<IconButton size={64} onClick={onClickMock} icon={menu_burger} />);
        expect(container).toMatchSnapshot();
    });
});
