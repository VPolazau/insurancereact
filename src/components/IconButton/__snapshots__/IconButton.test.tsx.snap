// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should match snapshot for custom size 1`] = `
<div>
  <button
    class="iconButtonBasic"
    data-testid="iconButtonComponent"
  >
    <img
      alt="iconButton"
      src="menu_burger.svg"
      style="height: 64px;"
    />
  </button>
</div>
`;

exports[`IconButton component should match snapshot for default size 1`] = `
<div>
  <button
    class="iconButtonBasic"
    data-testid="iconButtonComponent"
  >
    <img
      alt="iconButton"
      src="menu_burger.svg"
      style="height: 46px;"
    />
  </button>
</div>
`;
